### PR TITLE
feat: add button for delete row in header

### PR DESCRIPTION
**Pull request by harshithpabbati**
closes: #64 

Added a new delete row button in the header, displayed only when selected rows are greater than 0 and also checks if the table is editable.

### CODE REVIEW

File: src/types/grid.ts [hunk 1]
```diff
@@ -56,6 +56,10 @@ export type SupabaseGridProps = {
    * show add row button if available
    */
   onAddRow?: () => void;
+  /**
+   * show delete row button if available
+   */
+  onDeleteRow?: () => void;
```
Comment (phamhieu) [inline src/types/grid.ts:62]: Delete row should be handled by the grid. You can take a look [here](https://github.com/supabase/grid/blob/9d6d64b4466f1ed54a4ae61f98801ca5fb1ca7ec/src/components/menu/RowMenu.tsx#L21-L34)

**Review by phamhieu — CHANGES_REQUESTED** [review]

File: src/types/grid.ts (hunk 1 above)
Comment (harshithpabbati) [reply src/types/grid.ts]: @phamhieu is this what you suggested?

File: src/SupabaseGrid.tsx [hunk 2]
```diff
@@ -35,6 +35,7 @@ function cleanupProps(props: SupabaseGridProps) {
       ...props,
       onAddColumn: undefined,
       onAddRow: undefined,
+      onDeleteRow: undefined,
```
Comment (phamhieu) [inline src/SupabaseGrid.tsx:38]: we should remove this if it's unused

**Review by phamhieu — CHANGES_REQUESTED** [review]
Hi @harshithpabbati, I tested your changes. There are some issues
- On a readonly grid, when i select rows the delete button also shows which is wrong.
- Can we update the delete button style. More like `Insert row` with red background and recycleBin icon
- We also need a Yes/No confirm popup on delete.

**Comment by harshithpabbati** [comment]
Hi @phamhieu, sorry for the delay, I made the required changes 👍

**Comment by phamhieu** [comment]
Hi @harshithpabbati, I checked you latest changes. The delete button still shows on readonly mode, 

Screenshot 2021-07-19 at 3 40 27 PM

File: src/SupabaseGrid.tsx [hunk 3]
```diff
@@ -170,6 +170,7 @@ const SupabaseGridLayout = React.forwardRef<SupabaseGridRef, SupabaseGridProps>(
         <Header
           onAddRow={editable ? props.onAddRow : undefined}
           onAddColumn={editable ? props.onAddColumn : undefined}
+          editable={editable}
```
Comment (phamhieu) [inline src/SupabaseGrid.tsx:173]: you can get the grid editable param [from store](https://github.com/supabase/grid/blob/2c63a6fb398603b1dffa53fe2356b42810e10987/src/store/reducers/base.ts#L17). No need to pass it around.

**Review by phamhieu — CHANGES_REQUESTED** [review]

**Comment by harshithpabbati** [comment]
> Hi @harshithpabbati, I checked you latest changes. The delete button still shows on readonly mode,
> 
> Screenshot 2021-07-19 at 3 40 27 PM

Oh, I will look into it now

File: src/SupabaseGrid.tsx (hunk 3 above)
Comment (harshithpabbati) [reply src/SupabaseGrid.tsx:173]: Oh yeah, oops, will update it.

**Comment by harshithpabbati** [comment]
@phamhieu, I made the required changes 👍

**Comment by phamhieu** [comment]
lgtm, thank you @harshithpabbati

**Review by phamhieu — APPROVED** [review]

**Pull request merged**